hudi-utilities/src/test/java/org/apache/hudi/utilities/offlinejob/TestHoodieClusteringJob.java

```diff
@@ -92,6 +84,58 @@ public void testHoodieClusteringJobWithClean() throws Exception {
     HoodieOfflineJobTestBase.TestHelpers.assertNCleanCommits(1, tableBasePath, fs);
   }
 
+  @Test
+  public void testPurgePendingInstants() throws Exception {
+    String tableBasePath = basePath + "/purgePendingClustering";
+    Properties props = getPropertiesForKeyGen(true);
+    HoodieWriteConfig config = getWriteConfig(tableBasePath);
+    props.putAll(config.getProps());
+    Properties metaClientProps = HoodieTableMetaClient.withPropertyBuilder()
+        .setTableType(HoodieTableType.COPY_ON_WRITE)
+        .setPayloadClass(HoodieAvroPayload.class)
+        .fromProperties(props)
+        .build();
+
+    metaClient = HoodieTableMetaClient.initTableAndGetMetaClient(jsc.hadoopConfiguration(), tableBasePath, metaClientProps);
+    client = new SparkRDDWriteClient(context, config);
+
+    writeData(false, client.createNewInstantTime(), 100, true);
+    writeData(false, client.createNewInstantTime(), 100, true);
```

Comment:
`createNewInstantTime` is introduced in PR: https://github.com/apache/hudi/pull/9617. Can we avoid using this? @nsivabalan

Reply:
Is `HoodieActiveTimeline.createNewInstantTime()` working here?